src/serac/physics/tests/serac_solid_functional_finite_diff.cpp

```diff
@@ -156,6 +157,126 @@ TEST(SolidFunctionalFiniteDiff, FiniteDifference)
   }
 }
 
+TEST(solid_functional_finite_diff, finite_difference_shape)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  int serial_refinement   = 1;
+  int parallel_refinement = 0;
+
+  // Create DataStore
+  axom::sidre::DataStore datastore;
+  serac::StateManager::initialize(datastore, "solid_functional_parameterized_shape_sensitivities");
+
+  // Construct the appropriate dimension mesh and give it to the data store
+  std::string filename = SERAC_REPO_DIR "/data/meshes/beam-quad.mesh";
+
+  auto mesh = mesh::refineAndDistribute(buildMeshFromFile(filename), serial_refinement, parallel_refinement);
+  serac::StateManager::setMesh(std::move(mesh));
+
+  constexpr int p   = 1;
+  constexpr int dim = 2;
+
+  // Define a boundary attribute set
+  std::set<int> ess_bdr = {1};
+
+  double shape_displacement_value = 1.0;
+
+  // The nonlinear solver must have tight tolerances to ensure at least one Newton step occurs
+  SolverOptions options = {
+      DirectSolverOptions{},
+      NonlinearSolverOptions{.rel_tol = 1.0e-6, .abs_tol = 1.0e-12, .max_iter = 10, .print_level = 1}};
+
+  // Construct a functional-based solid solver
+  SolidFunctional<p, dim> solid_solver(options, GeometricNonlinearities::On, "solid_functional");
+
+  solid_mechanics::NeoHookean mat{1.0, 1.0, 1.0};
+  solid_solver.setMaterial(mat);
+
+  solid_solver.shapeDisplacement() = shape_displacement_value;
```

Comment:
At first glance I was confused why this was set with a `double`. Is this a rigid body translation? Also, perhaps it would be more consistent with other methods in solid functional to have a `setShapeDisplacement()` method?

Reply:
Yes - this is using the `mfem::Vector::operator=(double)` method. This brings up an important question though - would you rather have a set method and a const getter or direct access like this? @sethwatts @kswartz92 ?

Reply:
Any thoughts on this @kswartz92 @sethwatts ? I believe this is the last outstanding issue in this PR.

Reply:
I lean toward consistency with other methods (i.e., getter/setter) I will defer to @kswartz92 on this one.